### DISCUSSION TITLE
Extend type QueryKey

### DISCUSSION THREAD

**Discussion by IRediTOTO** [TanStack/query#4008] [Q&A]
Hi, we know QueryKey defined like this 
```
export declare type QueryKey = readonly unknown[];
```
Is there any way to extends it like this ?
```
export declare type QueryKey = readonly ("key1"|"key2")[];

```
Thank you for any answer :D

**Comment by TkDodo** [comment]
why do you need that? the query key is a generic that gets inferred from the key itself, or you can pass the generic to useQuery as well if you want ...
